marklogic-data-hub-central/ui/src/components/entities/create-edit-step-dialog/create-edit-step-dialog.tsx

```diff
@@ -149,9 +149,30 @@ const CreateEditStepDialog: React.FC<Props>  = (props) => {
   const confirmAction = () => {
     toggleConfirmModal(false);
     props.toggleModal(false);
+
+    resetTouchedValues()
   };
 
   const handleSubmit = async (event: { preventDefault: () => void; }) => {
+    if (!stepName) {
+      // missing name
+      setStepNameTouched(true);
+    }
+    if ((!collections && selectedSource === 'collection') || (!srcQuery && selectedSource !== 'collection')) {
+      // missing collection (if collection is selected)
+      setCollectionsTouched(true);
+    }
+    if ((!collections && selectedSource === 'collection') || (!srcQuery && selectedSource !== 'collection')) {
```

Comment:
It seems we want to check here if srcQuery has a value or not but used 'collections' again in the 'if' condition. 
The conditions in line 161 and 165 looks the same and redundant.

Reply:
Oops.  That was a mistake.  Glad you caught it.  Fixed now